e2e/pkg/dump/dump.go

```diff
@@ -46,6 +47,41 @@ func dumpResourceInAllNamespaces(ctx context.Context, resource client.ObjectList
 		return err
 	}
 
+	return dumpCleanUnstructuredOrResource(resource)
+}
+
+func dumpCleanUnstructuredOrResource(resource client.ObjectList) error {
+	if err := dumpCleanUnstructured(resource); err == nil {
+		return nil
+	}
+	return dumpResource(resource)
+}
+
+func dumpCleanUnstructured(resource client.ObjectList) error {
+	ur, err := runtime.DefaultUnstructuredConverter.ToUnstructured(resource)
```

Comment:
Why bother to go through the trouble of converting to unstructured to begin with?  It's going to get serialized to yaml in the end, and AIUI `yaml.Marshal` accepts structs, not just `map[string]interface{}`.